challenge-manager/edge-tracker/tracker.go

```diff
@@ -263,6 +273,14 @@ func (et *Tracker) Act(ctx context.Context) error {
 		return et.fsm.Do(edgeAwaitChallengeCompletion{})
 	// Edge tracker should add a subchallenge level zero leaf.
 	case EdgeAddingSubchallengeLeaf:
+		wasConfirmed, err := et.tryToConfirmEdge(ctx)
```

Comment:
I don't think this is accurate. We don't want to tryToConfirmEdge here, necessarily, same with the tryToConfirmEdge inside of edgeAtOneStepProof and bisecting. Instead, we want to await challenge completion if the path weight of this edge to its closest essential ancestor is >= 1 challenge period. if that's the case, then we move to edgeAwaitChallengeCompletion

Reply:
we'll need to add a new function that tells us the path weight from one edge to its closest essential ancestor

Reply:
Done.